wgpu-hal/src/lib.rs

```diff
@@ -839,6 +839,12 @@ pub struct NagaShader {
     pub info: naga::valid::ModuleInfo,
 }
 
+/// Shader input.
+pub enum ShaderInput<'a> {
+    Naga(NagaShader),
+    SpirV(Cow<'a, [u32]>),
```

Comment:
sorry, didn't notice this before. wgpu-hal's APIs do not use `Cow`. Let's make this one consistent and just accept `&[u32]` here